### ISSUE TITLE
Search can't find e.g. `__init__`

### ISSUE THREAD

**Issue by BioTurboNick** [JuliaDocs/Documenter.jl#2114]
https://docs.julialang.org/en/v1/search/?q=__init__

![image](https://user-images.githubusercontent.com/1438610/234297738-a59a825a-a888-4f2f-8cf0-fd99bf2691ff.png)

**Comment by mortenpi** [comment]
It's not surprising that this might fail. But how is `__init__` actually documented? I presume it's not a docstrings..

**Comment by BioTurboNick** [comment]
Ah, complicating matters is that `__init__` has some new documentation in 1.9. But it does have a docstring, in `base/docs/basedocs.jl`.

There are a few longer pages it shows up in that I'd expect to see, though, such as here: https://docs.julialang.org/en/v1/manual/modules/

**Comment by mortenpi** [comment]
Is the docstring included in the manual?

**Comment by KristofferC** [comment]
On dev it is at least: https://docs.julialang.org/en/v1.10-dev/base/base/#__init__

**Comment by BioTurboNick** [comment]
And that aside, `__module__` has an entry but isn't found by search either. It can be found by searching for `module`. So the underscores are maybe tripping up the index?

![image](https://user-images.githubusercontent.com/1438610/234593606-44588c5a-d889-48d5-8182-0185e8e78878.png)
![image](https://user-images.githubusercontent.com/1438610/234593725-a2c9a764-a35c-493c-953d-a373cc9503fa.png)

**Comment by mortenpi** [comment]
> So the underscores are maybe tripping up the index?

Yeah, we are probably stripping those characters somewhere, but then forget to do it in another place or something along those line.

**Comment by fredrikekre** [comment]
https://github.com/JuliaLang/julia/issues/45518

**Comment by timholy** [comment]
There's a package example that's much smaller than Julia's own docs, in case that helps debugging: https://github.com/JuliaLang/PrecompileTools.jl/issues/34#issuecomment-1925275043. You'll see I created a gist with the contents of the search index, in case it helps debug this.

**Comment by nsajko** [comment]
On https://docs.julialang.org/en/v1.12-dev/, the search also can't find `.=`, `->` or `::`, even though they're now included in the docs.

xref JuliaLang/julia#36290

**Comment by timholy** [comment]
Note that in https://github.com/JuliaLang/PrecompileTools.jl/issues/34#issuecomment-1925275043 I've found that:
- the target string *does* appear in the `search_index.js` file
- there are behaviors, such as changing results simply by adding `.jl` to the end of the string, that make one think it's the search logic that's at fault

This is a pretty big issue. I don't speak javascript, unfortunately.

**Comment by fingolfin** [comment]
The title issue (searching for `__init__`) is resolved, also similar ones like searching for `__module__`.

That searching for symbols like `.=` or `&&` often doesn't work is also reported in issue #1457.

Perhaps the title of this issue should be adjusted, and/or this issue be consolidated with #1457.

Problematic search strings mentioned in the two issues are:
- `.=`
- `->`
- `::`
- `&&`
- `^`

But there are many more, e.g. it might be useful to search for `#` or `+` in the main Julia docs...
